data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/config/GetDepositContract.java

```diff
@@ -13,38 +13,62 @@
 
 package tech.pegasys.teku.beaconrestapi.handlers.v1.config;
 
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_INTERNAL_SERVER_ERROR;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_INTERNAL_ERROR;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_CONFIG;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.UINT64_TYPE;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.javalin.http.Context;
-import io.javalin.http.Handler;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
-import java.util.Optional;
+import java.util.function.Function;
+import org.jetbrains.annotations.NotNull;
 import tech.pegasys.teku.api.ConfigProvider;
 import tech.pegasys.teku.api.response.v1.config.GetDepositContractResponse;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.spec.datastructures.eth1.Eth1Address;
 
-public class GetDepositContract implements Handler {
+public class GetDepositContract extends MigratingEndpointAdapter {
   public static final String ROUTE = "/eth/v1/config/deposit_contract";
-  private Optional<String> depositContractResponse;
-  private final String depositContractAddress;
-  private final JsonProvider jsonProvider;
+  private final Eth1Address depositContractAddress;
   private final ConfigProvider configProvider;
 
+  private static final SerializableTypeDefinition<DepositContractData> DEPOSIT_CONTRACT_TYPE =
+      SerializableTypeDefinition.object(DepositContractData.class)
+          .withField("chain_id", UINT64_TYPE, DepositContractData::getChainId)
```

Comment:
nit: we might want to set description for this specific uint64... i think its something like
`UINT64_TYPE.withDescription("Id of Eth1 chain on which contract is deployed.")`